sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -290,7 +309,78 @@ exception is caught and handled appropriately.
 _{endnote}_]
 |====
 
-=== New properties
+=== New properties for the `create_kernel_bundle_from_source` function
+
+This extension adds the following properties, which can be used in conjunction
+with the `create_kernel_bundle_from_source` function that is defined above:
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct include_files {
+  std::vector<std::string> names;
+  std::vector<std::string> contents;
+  include_files(const std::string &name, const std::string &content);  (1)
+  include_files(const std::vector<std::string> &names,                 (2)
+                const std::vector<std::string> &contents);
+};
+using include_files_key = include_files;
+
+template<>
+struct is_property_key<include_files_key> : std::true_type {};
+
+} // namespace sycl::ext::oneapi::experimental
+----
+!====
+
+This property provides the name and content of include files that can be
+referenced from the source code in the `source` parameter to
+`create_kernel_bundle_from_source`.
+The property conceptually contains a collection of (_Name_, _Content_) pairs,
+where both _Name_ and _Content_ are strings.
+The _Name_ is the name of an include file and the _Content_ is the content of
+that include file.
+
+When the source language is `source_language::sycl`, the source code can have
+`#include` statements where the name and content of the include file is
+defined by this property.
```

Comment:
I agree that making it a template argument would allow compile-time error checking here.  However, I think it is valuable to allow applications to pass `lang` as a runtime value, and I don't want to require them to code a dispatch table in this case.  Therefore, I'm more inclined to diagnose a runtime error if a source language does not support one of the properties.

Reply:
Okey, sounds good to me. Now, should we add a _Throws_ clause to `create_kernel_bundle_from_source` in case `lang` does not support any of the input properties or just state they'll be omitted?

Reply:
Yup, I just ran out of time yesterday.  Done now in 1c3f01b308562cc2d1c9253ec0cd2b1bde5b6bfc.